packs/tfe_fdo_nomad/templates/tfe.nomad.tpl

```diff
@@ -0,0 +1,220 @@
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: MPL-2.0
+
+job [[ .tfe_fdo_nomad.job_name | quote ]] {
+  datacenters = [[ .tfe_fdo_nomad.datacenters  | toStringList ]]
+  namespace   = [[ .tfe_fdo_nomad.tfe_namespace | quote ]]
+  type        = "service"
+
+  group "tfe-group" {
+    count = [[ .tfe_fdo_nomad.tfe_group_count ]]
+    
+    spread {
+      attribute = "${node.unique.id}"
+    }
+
+    restart {
+      attempts = 3
+      delay    = "60s"
+      interval = "10m"
+      mode     = "fail"
+    }
+
+    update {
+      max_parallel      = 1
+      min_healthy_time  = "30s"
+      healthy_deadline  = "12m"
+      progress_deadline = "15m"
+      health_check      = "checks"
+    }
+
+
+    network {
+      port "tfe" {
+        static = [[ .tfe_fdo_nomad.tfe_port ]]
+      }
+      port "http" {
+        static = [[ .tfe_fdo_nomad.tfe_http_port ]]
+      }
+      port "vault" {
+        static = [[ .tfe_fdo_nomad.tfe_vault_cluster_port ]]
+      }
+    }
+
+    service {
+      name     = [[ .tfe_fdo_nomad.tfe_service_name | quote ]]
+      port     = "tfe"
+      provider = "nomad"
+
+      check {
+        type     = "http"
+        port     = "http"
+        path     = "/_health_check"
+        interval = "5s"
+        timeout  = "2s"
```

Comment:
These should be configurable because they could be a bit too aggressive for certain customers / under heavy load.